[CI] ClientYamlTestSuiteIT test {yaml=search.vectors/41_knn_search_bbq_hnsw/Vector rescoring has same scoring as exact search for kNN section} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #11818 / almalinux-8-aarch64_checkrestcompat_platform-support-arm](https://gradle-enterprise.elastic.co/s/ntjbhxx4hjomo)
- [elasticsearch-periodic-platform-support #11818 / ubuntu-2404-aarch64_checkrestcompat_platform-support-arm](https://gradle-enterprise.elastic.co/s/eviosuxxieub2)
- [elasticsearch-periodic-platform-support #11818 / windows-2025_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/lxe36jlgecgtc)
- [elasticsearch-periodic-platform-support #11818 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/l6pdht6xnfpb6)
- [elasticsearch-periodic-platform-support #11818 / windows-2025_checkrestcompat_platform-support-windows](https://gradle-enterprise.elastic.co/s/ngvenxvgfzzoq)
- [elasticsearch-periodic-platform-support #11818 / windows-2022_checkrestcompat_platform-support-windows](https://gradle-enterprise.elastic.co/s/lul6ypbo3rdwq)
- [elasticsearch-periodic-platform-support #11818 / ubuntu-2404-aarch64_checkrestcompat_platform-support-arm](https://gradle-enterprise.elastic.co/s/jznk7hsilgidm)
- [elasticsearch-periodic-platform-support #11818 / almalinux-8-aarch64_checkrestcompat_platform-support-arm](https://gradle-enterprise.elastic.co/s/pj56dxmzfra4a)
- [elasticsearch-periodic-platform-support #11818 / windows-2025_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/zkso4vs6523bq)
- [elasticsearch-periodic-platform-support #11818 / windows-2022_checkrestcompat_platform-support-windows](https://gradle-enterprise.elastic.co/s/244vyo4uacgpi)

**Reproduction Line:**
```
./gradlew ":rest-api-spec:yamlRestCompatTest" --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT" -Dtests.method="test {yaml=search.vectors/41_knn_search_bbq_hnsw/Vector rescoring has same scoring as exact search for kNN section}" -Dtests.seed=2C90DB2ED34D9ED2 -Dtests.locale=ta-SG -Dtests.timezone=Africa/Nouakchott -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.test.rest.ClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Byaml%3Dsearch.vectors%2F41_knn_search_bbq_hnsw%2FVector%20rescoring%20has%20same%20scoring%20as%20exact%20search%20for%20kNN%20section%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: java.lang.AssertionError: 
Expected: <6.0841703>
     but: was <6.08417>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {yaml=search.vectors/41_knn_search_bbq_hnsw/Vector rescoring has same scoring as exact search for kNN section}
- [main] 2 consecutive failures in step almalinux-8-aarch64_checkrestcompat_platform-support-arm
- [main] 2 consecutive failures in step ubuntu-2404-aarch64_checkrestcompat_platform-support-arm
- [main] 2 consecutive failures in step windows-2025_checkpart1_platform-support-windows
- [main] 2 consecutive failures in step windows-2022_checkpart1_platform-support-windows
- [main] 2 consecutive failures in step windows-2025_checkrestcompat_platform-support-windows
- [main] 2 consecutive failures in step windows-2022_checkrestcompat_platform-support-windows
- [main] 13 failures in test test {yaml=search.vectors/41_knn_search_bbq_hnsw/Vector rescoring has same scoring as exact search for kNN section} (2.6% fail rate in 502 executions)
- [main] 2 failures in step almalinux-8-aarch64_checkrestcompat_platform-support-arm (66.7% fail rate in 3 executions)
- [main] 2 failures in step ubuntu-2404-aarch64_checkrestcompat_platform-support-arm (66.7% fail rate in 3 executions)
- [main] 2 failures in step windows-2025_checkpart1_platform-support-windows (66.7% fail rate in 3 executions)
- [main] 2 failures in step windows-2022_checkpart1_platform-support-windows (66.7% fail rate in 3 executions)
- [main] 2 failures in step windows-2025_checkrestcompat_platform-support-windows (66.7% fail rate in 3 executions)
- [main] 2 failures in step windows-2022_checkrestcompat_platform-support-windows (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -356,9 +356,6 @@ tests:
 - class: org.elasticsearch.xpack.ccr.CcrTimeSeriesDataStreamsIT
   method: testCrossClusterReplicationForTSDBWithSyntheticId
   issue: https://github.com/elastic/elasticsearch/issues/141796
-- class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
-  method: test {yaml=search.vectors/41_knn_search_bbq_hnsw/Vector rescoring has same scoring as exact search for kNN section}
-  issue: https://github.com/elastic/elasticsearch/issues/141842
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=esql/50_index_patterns/disjoint_mappings}
   issue: https://github.com/elastic/elasticsearch/issues/141856

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -362,9 +362,6 @@ tests:
 - class: org.elasticsearch.index.codec.vectors.diskbbq.PreconditionerTests
   method: testRandomProviderConfigurations
   issue: https://github.com/elastic/elasticsearch/issues/141748
-- class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
-  method: test {yaml=search.vectors/41_knn_search_bbq_hnsw/Vector rescoring has same scoring as exact search for kNN section}
-  issue: https://github.com/elastic/elasticsearch/issues/141842
 
 # Examples:
 #

--- a/rest-api-spec/build.gradle
+++ b/rest-api-spec/build.gradle
@@ -125,4 +125,5 @@ tasks.named("yamlRestCompatTestTransform").configure ({ task ->
   // Expected deprecation warning to compat yaml tests:
   task.addAllowedWarningRegex("Use of the \\[max_size\\] rollover condition has been deprecated in favour of the \\[max_primary_shard_size\\] condition and will be removed in a later version")
   task.skipTest("search.vectors/42_knn_search_bbq_flat/Vector rescoring has same scoring as exact search for kNN section", "scores have changed slightly with native implementations")
+  task.skipTest("search.vectors/41_knn_search_bbq_hnsw/Vector rescoring has same scoring as exact search for kNN section", "scores have changed slightly with native implementations")
 })

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/41_knn_search_bbq_hnsw.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/41_knn_search_bbq_hnsw.yml
@@ -111,7 +111,7 @@ setup:
 "Vector rescoring has same scoring as exact search for kNN section":
   - requires:
       reason: 'Quantized vector rescoring is required'
-      test_runner_features: [capabilities]
+      test_runner_features: [capabilities, close_to]
       capabilities:
         - method: GET
           path: /_search
@@ -172,9 +172,9 @@ setup:
 
   # Compare scores as hit IDs may change depending on how things are distributed
   - match: { hits.total: 3 }
-  - match: { hits.hits.0._score: $rescore_score0 }
-  - match: { hits.hits.1._score: $rescore_score1 }
-  - match: { hits.hits.2._score: $rescore_score2 }
+  - close_to: { hits.hits.0._score: { value: $rescore_score0, error: 0.00001 }}
+  - close_to: { hits.hits.1._score: { value: $rescore_score1, error: 0.00001 }}
+  - close_to: { hits.hits.2._score: { value: $rescore_score2, error: 0.00001 }}
 
 ---
 "Test bad quantization parameters":

